lib/providers/services/http_service.dart

```diff
@@ -236,7 +236,34 @@ class HttpService {
     } else {
       print("Cannot fetch balance...");
 
-      return BalanceModel("0.0000 SEEDS");
+      return BalanceModel("0.0000 SEEDS", true);
+    }
+  }
+
+    Future<RateModel> getUSDRate() async {
+    print("[http] get seeds rate USD");
+
+    if (mockResponse == true) {
+      return HttpMockResponse.rate;
+    }
+
+    final String rateURL = '$baseURL/v1/chain/get_table_rows';
+
+    String request =
+        '{"json":true,"code":"tlosto.seeds","scope":"tlosto.seeds","table":"price"}';
+    
+    Map<String, String> headers = {"Content-type": "application/json"};
+
+    Response res = await post(rateURL, headers: headers, body: request);
```

Comment:
We should also have a timeout on this. Say 5 seconds or something. Not sure what the default is, but it wouldn't surprise me if it's more like 30 seconds.